test/libqlever/QleverTest.cpp

```diff
@@ -12,6 +12,118 @@
 using namespace qlever;
 using namespace testing;
 
+// _____________________________________________________________________________
+TEST(LibQlever, buildIndexAndRunQuery) {
+  std::string filename = "testIndexForLibQlever.ttl";
+  {
+    auto ofs = ad_utility::makeOfstream(filename);
+    ofs << "<s> <p> <o>. <s2> <p> \"kartoffel und salat\".";
+  }
+
+  IndexBuilderConfig c;
+  c.inputFiles_.emplace_back(filename, Filetype::Turtle);
+  c.baseName_ = "testIndexForLibQlever";
+
+  // Test the activation of the memory limit
+  c.memoryLimit_ = ad_utility::MemorySize::bytes(0);
+  EXPECT_ANY_THROW(Qlever::buildIndex(c));
+
+  c.memoryLimit_ = std::nullopt;
+
+  c.parserBufferSize_ = ad_utility::MemorySize::bytes(0);
+  EXPECT_ANY_THROW(Qlever::buildIndex(c));
+
+  c.parserBufferSize_ = std::nullopt;
+  EXPECT_NO_THROW(Qlever::buildIndex(c));
+  {
+    EngineConfig ec{c};
+    Qlever engine{ec};
+    // Run a simple query.
+    auto res = engine.query("SELECT ?s WHERE { ?s <p> <o> }",
+                            ad_utility::MediaType::tsv);
+    EXPECT_EQ(res, "?s\n<s>\n");
+
+    // Run with a different media type
+    res = engine.query("SELECT * WHERE { <s> <p> ?o }",
+                       ad_utility::MediaType::csv);
+    EXPECT_EQ(res, "o\no\n");
+
+    // Separately run the planning and the query.
+    auto plan = engine.parseAndPlanQuery("SELECT * WHERE { <s> <p> ?o }");
+    res = engine.query(plan, ad_utility::MediaType::csv);
+    EXPECT_EQ(res, "o\no\n");
+  }
+
+  c.addWordsFromLiterals_ = true;
+
+  // Note: Currently the `addWordsFromLiterals` feature is broken, but
+  // @flixtastic has a fix for this.
+  EXPECT_NO_THROW(Qlever::buildIndex(c));
+  EngineConfig ec{c};
+  ec.loadTextIndex_ = true;
+  Qlever engine{ec};
+}
+
+// _____________________________________________________________________________
+TEST(LibQlever, fulltextIndex) {
+  std::string filename = "testIndexForLibQleverFulltext.ttl";
+  std::string wordsfileName = "testIndexForLibQleverFulltext.words";
+  std::string docsFileName = "testIndexForLibQleverFulltext.docs";
+  {
+    auto ofs = ad_utility::makeOfstream(filename);
+    ofs << "<s> <p> <o>. <s2> <p> \"kartoffel und salat\".";
+    auto words = ad_utility::makeOfstream(wordsfileName);
+    words << "kartoffel\t0\t13\t1\n<s>\t1\t13\t1\n";
+    auto docs = ad_utility::makeOfstream(docsFileName);
+    docs << "13\tKartoffeln sind ein schönes Gemüse";
```

Comment:
To be honest, for robustness you could append `🥔` at the end to make sure non-ascii characters are not resulting in exceptions.

Also there's `🥗` for the turtle file for the fun of it.